### PR TITLE
Update fuzzing infra to build and use the shared framework

### LINKED ISSUE #107204: Fuzzing workflow is not friendly to rebuild and incremental changes.
I noticed that the fuzzing workflow runs as a self-contained app, which means that it requires a full repo build and will have incremental build challenges since self-contained apps rely on restoring nuget packages.

I see that as a result of this, the fuzzing infrastructure defines it's own NuGet.config with a local packages cache and pointers to the output packages.  That means any time someone rebuilds they need to delete that cache and rebuild the entire repo to produce those packages.

Can you make it a framework dependent app instead, similar to how all our regular tests work, and use the test shared framework?

That's optimized for local builds and incremental changes - rebuilding libraries will update the test sharedframework without any extra steps.

@MihaZupan

### PR DESCRIPTION
**Pull request by buyaa-n**
Fuzzing infra is currently broken because of https://github.com/dotnet/runtime/pull/106599, because it is partially integrated with the repo workflow. In order to fix this failure, we need to integrate it completely, that means need to fix https://github.com/dotnet/runtime/issues/107204:
- Make the fuzzing solution a framework dependent app and use test shared framework 
- Fixed and/or suppressed all findings of different analyzer runs
- Updated the README.md to apply the changes made
- Ended up removing `nuget.config` as it became same as the repo config in the root, also removed `Directory.Build.targets` that was empty

Further `MihuBot` need to be updated as needed, might also need to update `onefuzz` deployment scripts CC @MihaZupan

### CODE REVIEW

File: src/libraries/Fuzzing/README.md [hunk 1]
```diff
@@ -18,28 +18,34 @@ Useful links:
 
 ### Prerequisites
 
-Build the runtime with the following arguments:
+Build the runtime if haven't already:
 ```cmd
-./build.cmd clr+libs+packs+host -rc Checked -c Debug
+./build.cmd clr+libs -rc release
```
Comment (MihaZupan) [inline src/libraries/Fuzzing/README.md:23]: Why this change?

Using checked+debug was intentional so that we run with all debug asserts enabled, including in corelib

Comment (buyaa-n) [reply src/libraries/Fuzzing/README.md:23]: Updated it mainly because before `+packs+host` was required to be built, but not now, just tried to show now it can run with a normal build, though your point to use checked for runtime by default makes sense, I can change that

Comment (MihaZupan) [reply src/libraries/Fuzzing/README.md:23]: Not having to build `packs+host` will be nice to reduce build times, but I would keep it on checked+debug.

Comment (MihaZupan) [reply src/libraries/Fuzzing/README.md:23]: though I suppose it doesn't really matter as much anymore, the build works no matter which one you build locally.

File: src/libraries/Fuzzing/README.md [hunk 2]
```diff
@@ -18,28 +18,34 @@ Useful links:
 
 ### Prerequisites
 
-Build the runtime with the following arguments:
+Build the runtime if haven't already:
```
Comment (MihaZupan) [inline src/libraries/Fuzzing/README.md:21]:
```suggestion
Build the runtime if you haven't already:
```

File: src/libraries/Fuzzing/README.md [hunk 3]
```diff
@@ -18,28 +18,34 @@ Useful links:
...
 and install the SharpFuzz command line tool:
 ```cmd
 dotnet tool install --global SharpFuzz.CommandLine
 ```
 
 > [!TIP]
-> The project uses a checked runtime + debug libraries configuration by default.
-> If you want to use a different configuration, make sure to also adjust the artifact paths in `nuget.config`.
+> The project uses a release runtime + debug libraries configuration by default.
+> If you want to use a different configuration, make sure to also adjust the paths in `run.bat`.
```
Comment (MihaZupan) [inline src/libraries/Fuzzing/README.md:33]:
```suggestion
> Automated fuzzing runs use a checked runtime + debug libraries configuration by default.
> You can use any configuration locally, but `checked` is recommended when testing changes in `System.Private.CoreLib`
```

File: src/libraries/Fuzzing/README.md [hunk 4]
```diff
@@ -18,28 +18,34 @@ Useful links:
...
 
 ### Fuzzing locally
 
-The `prepare-onefuzz` command will create separate directories for each fuzzing target, instrument the relevant assemblies, and generate a helper script for running them locally.
-Note that this command must be ran on the published artifacts (won't work with `dotnet run`).
+Build the `DotnetFuzzing` fuzzing project, the project is self contained so it will produce `DotnetFuzzing.exe` and all libraries needed for run into the output.
```
Comment (MihaZupan) [inline src/libraries/Fuzzing/README.md:37]:
```suggestion
Build the `DotnetFuzzing` fuzzing project. It is self-contained, so it will produce `DotnetFuzzing.exe` along with a copy of all required libraries.
```

File: src/libraries/Fuzzing/README.md [hunk 5]
```diff
@@ -18,28 +18,34 @@ Useful links:
...
 
 ```cmd
 cd src/libraries/Fuzzing/DotnetFuzzing
 
-dotnet publish -o publish; publish/DotnetFuzzing.exe prepare-onefuzz deployment
+dotnet build
+```
+
+Now can `run run.bat` that will finds the path to the `DotnetFuzzing.exe` and run it with required arguments `\path\DotnetFuzzing.exe prepare-onefuzz deployment`. The `prepare-onefuzz` command will create separate directories for each fuzzing target, instrument the relevant assemblies, and generate a helper script for running them locally.
```
Comment (MihaZupan) [inline src/libraries/Fuzzing/README.md:45]:
```suggestion
Now you can run `run.bat`, which will create separate directories for each fuzzing target, instrument the relevant assemblies, and generate a helper script for running them locally.
```

**Review by MihaZupan — APPROVED** [review]
Thank you

**Comment by MihaZupan** [comment]
The bot's also working again now https://github.com/MihuBot/runtime-utils/commit/5466698d915fbc21a68e5182015deb09954dd94c

File: src/libraries/Common/tests/TestUtilities/System/Buffers/BoundedMemory.Windows.cs [hunk 6]
```diff
@@ -30,12 +30,15 @@ private static WindowsImplementation<T> AllocateWithoutDataPopulationWindows<T>(
...
 
             // Reserve and commit the entire range as NOACCESS.
-
-            VirtualAllocHandle handle = VirtualAllocHandle.Allocate(
-                lpAddress: IntPtr.Zero,
-                dwSize: (IntPtr)totalBytesToAllocate /* cast throws OverflowException if out of range */,
-                flAllocationType: VirtualAllocAllocationType.MEM_RESERVE | VirtualAllocAllocationType.MEM_COMMIT,
-                flProtect: VirtualAllocProtection.PAGE_NOACCESS);
+            VirtualAllocHandle handle;
+            checked
```
[the remark this answers is not in the document]
Comment (buyaa-n) [reply src/libraries/Common/tests/TestUtilities/System/Buffers/BoundedMemory.Windows.cs:34]: Not due to a bug, after changing the project build with current runtime settings many analyzer diagnostics found that are fixed or suppressed with the PR.

[CA2020](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/CA2020)  `Starting with .NET 7 the explicit conversion '(IntPtr)Int64' will not throw when overflowing in an unchecked context. Wrap the expression with a 'checked' statement to restore the .NET 6 behavior.` 

Flagged at row 38: `dwSize: (IntPtr)totalBytesToAllocate /* cast throws OverflowException if out of range */,`

Given that this type is for testing, and also base on the comment I assumed it is important to restore the overflowing behavior

[the remark this answers is not in the document]
Comment (buyaa-n) [reply src/libraries/Common/tests/TestUtilities/System/Buffers/BoundedMemory.Windows.cs:34]: I suppressed have all other warnings for this `BoundedMemory` type, now after moving `<IsTestSupportProject>true</IsTestSupportProject>` property above importing parent `Directory.Build.props` this type is not being analyzed anymore, same as all other test types. I can revert this and other suppressions with my other PR

File: src/libraries/Fuzzing/README.md [hunk 7]
```diff
@@ -18,28 +18,36 @@ Useful links:
...
+Build the `DotnetFuzzing` fuzzing project. It is self-contained, so it will produce `DotnetFuzzing.exe` along with a copy of all required libraries.
 
 ```cmd
 cd src/libraries/Fuzzing/DotnetFuzzing
 
-dotnet publish -o publish; publish/DotnetFuzzing.exe prepare-onefuzz deployment
+dotnet build
+```
+
+Now you can run `run.bat`, which will create separate directories for each fuzzing target, instrument the relevant assemblies, and generate a helper script for running them locally.
```
[the remark this answers is not in the document]
Comment (buyaa-n) [reply src/libraries/Fuzzing/README.md:46]: I can update with my other PR
